pandoc/README.md

```diff
@@ -0,0 +1,86 @@
+<tr>
+<td><b>Input directories</b></td>
+<td><code>dotty/docs/docs/&lt;project&gt;/</code><br/><code>dotty/pandoc/data/</code></td>
+<td><code>*.md</code> files<br/><a href="https://pandoc.org/">Pandoc</a> data files</td>
+</tr>
+<tr>
+<td><b>Output directory</b></td>
+<td><code>dotty/out/pandoc/</code></td>
+<td><code>scala3_<i>&lt;project&gt;</i>.pdf</code><sup>(*)</sup></td>
+</tr>
+</table>
+
+<span style="margin-left:10px;font-size:90%;"><sup>(*)</sup> *`<project>`* is one of [`contributing`](https://github.com/lampepfl/dotty/tree/master/docs/docs/contributing), [`internals`](https://github.com/lampepfl/dotty/tree/master/docs/docs/internals), [`reference`](https://github.com/lampepfl/dotty/tree/master/docs/docs/reference) or [`usage`](https://github.com/lampepfl/dotty/tree/master/docs/docs/usage).</span>
+
+The generated PDF file is more elaborated than its sibling HTML version; unlike the online [*Scala 3 reference*][scala3_reference] documentation available on the [Scala 3 documentation](https://docs.scala-lang.org/scala3/) webpage, the `scala3_reference.pdf` document :
+- starts with a *title page* directly followed by a *table of contents* and end with an *appendix*.
+- gathers all the "*More details*" sections in the appendix.
+
+> See document [`PROJECT.md`](docs/PROJECT.md) for further information, e.g. project organisation.
+
+## <span id="dependencies">Project dependencies</span> <sup style="font-size:60%;">[**&#9650;**](#top)</sup>
+
+This project depends on the following software :
+
+- [Pandoc 2](https://github.com/jgm/pandoc/releases) <sup id="anchor_01">[[1]](#footnote_01)</sup> *([release notes](https://pandoc.org/releases.html))*
+- [TeX Live 2020](https://tug.org/texlive/) <sup id="anchor_02">[[2]](#footnote_02)</sup> *([release notes](http://www.tug.org/texlive/doc/texlive-en/texlive-en.html#x1-880009.2))*
+
+> See documents [`PANDOC.md`](docs/PANDOC.md) and [`TEXLIVE.md`](docs/TEXLIVE.md) for product specific information, e.g. software installation.
+
+One may also install the following software:
+
+- [Docker Desktop 3](https://docs.docker.com/get-docker/) <sup id="anchor_03">[[3]](#footnote_03)</sup> *([release notes](https://docs.docker.com/release-notes/))*
+
+> See document [`DOCKER.md`](docs/DOCKER.md) for further information, e.g. `Dockerfile` usage.
+
+## <span id="commands">Build commands</span> <sup style="font-size:60%;">[**&#9650;**](#top)</sup>
+
+We provide two commands for MacOS/Ubuntu as well as their equivalents for MS Windows :
+- for *interactive users* :<br/>[`build {<option>|<subcommand>}`](./build) (resp. [`build.bat`](./build.bat) on MS Windows)<br/><span style="font-size:80%;">(option `-help` displays the available options/subcommands).</span>
+- for *automated tasks* (primarily, but not exclusively) :<br/>[`md2pdf [<project>]`](./md2pdf) (resp. [`md2pdf.bat`](./md2pdf.bat) on MS Windows)<br/><span style="font-size:80%;">(default value for `<project>` : `reference`).</span>
```

Comment:
I've wondering, instead of having two commands `build` and `md2pdf`, can we have the same command for both local machines and cloud CI, assuming that both will use a published docker image?

Reply:
Your suggestion is right.

The existence of both commands `build` and `md2pdf` stems in part from my own background as a long-standing developer. Let me try to advocate the case (*for fun*) : 
- as a preamble the existance of both scripts is not directly linked to Docker.
- [`build`](https://github.com/michelou/dotty/blob/scala3-pandoc/pandoc/build) is a user-friendly wrapper script; it calls `md2pdf` but it also
   - provides some options (e.g. `-verbose` and `-debug`) and some subcommands (e.g. `clean` and `view` on MS Windows); I typically run command `build -timer -verbose clean view` to generate file [`scala3_reference.pdf`](https://github.com/michelou/dotty-examples/blob/master/docs/scala3_reference.pdf) on my Win10 laptop.
   - provides conditional compilation; the output file (e.g. [`scala3_reference.pdf`](https://github.com/michelou/dotty-examples/blob/master/docs/scala3_reference.pdf)) is generated depending on input file timestamps.
-  [`md2pdf`](https://github.com/michelou/dotty/blob/scala3-pandoc/pandoc/md2pdf) is the minimal script for running [`pandoc`](https://pandoc.org/MANUAL.html) with the appropriate settings.

In the context of the Dotty project, we can indeed have *only one*  shell script :  [`md2pdf`](https://github.com/michelou/dotty/blob/scala3-pandoc/pandoc/md2pdf) (and may be [`md2pdf.bat`](https://github.com/michelou/dotty/blob/scala3-pandoc/pandoc/md2pdf.bat)).

**NB,** I could even omit several ~~type checks~~ environment checks in`md2pdf` assuming users can help themselves in case of errors.